cmd/saml2aws/commands/login.go

```diff
@@ -79,10 +82,30 @@ func Login(loginFlags *flags.LoginExecFlags) error {
 
 	log.Printf("Authenticating as %s ...", loginDetails.Username)
 
-	samlAssertion, err := provider.Authenticate(loginDetails)
-	if err != nil {
-		return errors.Wrap(err, "error authenticating to IdP")
+	var samlAssertion string
+	if account.SAMLCache {
+		if valid, err := cacheProvider.IsValid(); valid {
```

Comment:
Your ignoring the error here, if there is an error you should probably log it and stop

Reply:
good call. will fix

Reply:
refactored `IsValid` to returning a bool only